bluetooth: host: Configurable HCI stack sizes

mtpr-ot opened the pull request:
Made BT_HCI_TX_STACK_SIZE configurable and added BT_HCI_ECC_STACK_SIZE.
These changes are necessary to allow vendors to adjust for other achitectures.

Signed-off-by: Morten Priess <mtpr@oticon.com>

jhedberg reviewed (changes requested):
It seems to me this is more related to out-of-tree platforms rather than nRF, correct? Since for in-tree HCI drivers (or controllers) we have the per-driver defaults in Kconfig.

aescolar commented on subsys/bluetooth/host/Kconfig:
```diff
@@ -81,6 +81,10 @@ config BT_HCI_TX_STACK_SIZE
 	  Stack size needed for executing bt_send with specified driver
 
+config BT_HCI_ECC_STACK_SIZE
+	int "HCI ECC thread stack size" if !SOC_COMPATIBLE_NRF
```
@jhedberg that `if !` just makes the prompt conditional. So the option is not user configurable for NRF platforms (meaning it is mostly just as before, except that that constant is moved to a hidden kconfig option).
The option only appears and is configurable for other platforms where, due to architecture differences, you may need to tune that stack size.

jhedberg replied:
@aescolar so you're saying there are architectures where the same TinyCrypt APIs consume more stack than what's currently configured?  Which architectures are those? If the difference isn't more than 100 bytes or so I'd just increase the global default. If it's more then couldn't we just have architecture-specific defaults for this value?

mtpr-ot replied:
Currently we use 1080 bytes in a test run - this might not be max. We assert when passing 90% usage, just to make sure we have a margin. We would need something like 1200-1300, and _can_ disable the 90% limit for this stack, but it seems close. It seems like an "expensive" solution to generally increase by 200 bytes...

jhedberg commented:
I understand more or less the need & desire to make these options user configurable. My biggest issue with this PR is still the usage of nRF as a decision point. We have many boards in the tree that have Bluetooth support but don't use the Zephyr native controller, e.g. the ones from ST with a SPI-based controller, the ST one with shared-memory (CONFIG_BT_STM32_IPM), various ones from 96Boards, and even Nordic's nRF91 which uses UART HCI transport between a Cortex M33 and an nRF52 (Cortex 4), to name a few. It seems arbitrary that all these boards should suddenly have these Kconfig options user-configurable but nRF-based ones shouldn't.

mtpr-ot commented:
> ... It seems arbitrary that all these boards should suddenly have these Kconfig options user-configurable but nRF-based ones shouldn't.

I agree. Would it be OK then to have them always configurable, and add a "warning" in the help message?

jhedberg commented:
> I agree. Would it be OK then to have them always configurable, and add a "warning" in the help message?

Yes, I think that's reasonable, i.e. something along the lines of "Don't touch this unless you know what you're doing"

mtpr-ot committed:
bluetooth: host: Configurable HCI stack sizes

Made BT_HCI_TX_STACK_SIZE configurable and added BT_HCI_ECC_STACK_SIZE.
These changes are necessary to allow vendors to adjust for other
achitectures.

Signed-off-by: Morten Priess <mtpr@oticon.com>
